_connect-content/guides/api/select-tables-and-fields.md

```diff
@@ -0,0 +1,309 @@
+
+    substeps:
+      - title: "Get the source's last connection check"
+        anchor: "get-sources-connection-check"
+        content: |
+          {% assign right-bracket = "}" %}
+
+          To view the results of the source's [last connection check]({{ site.data.connect.api.section | flatify | prepend: site.baseurl | append: site.data.connect.core-objects.connection-checks.object }}), make a request to `GET {{ site.data.connect.core-objects.connection-checks.get-source.name | flatify }}`, replacing `{source_id}` with the source's ID:
+
+          ```json
+          curl -X GET {{ site.data.connect.api.base-url | strip_newlines }}{{ site.data.connect.core-objects.connection-checks.get-source.name | flatify | replace: "{source_id",source-id | remove: right-bracket | strip_newlines }}
+               -H 'Content-Type: application/json' \
+               -H 'Authorization: Bearer <API_TOKEN>'
+          ```
+
+          A successful connection check and discovery will have a `status` of `succeeded` and a `discovery_exit_status` of `0`:
+
+          ```json
+          {{ site.data.connect.code-examples.connection-checks.successful | rstrip }}
+          ```
+
+          When the connection check completes, the source's `current_step` will advance to `field_selection`.
+        
+      - title: "Verify the current connection step"
+        anchor: "verify-current-connection-step"
+        content: |
+          Next, you'll verify that the source has advanced to the `field_selection` step. This step indicates that available streams and fields can be selected for replication.
+
+          To get the source's `current_step`, make a request to `GET {{ site.data.connect.core-objects.sources.retrieve.name | flatify }}`, replacing `{source_id}` with the source's ID:
+
+          ```json
+          curl -X GET {{ site.data.connect.api.base-url | strip_newlines }}{{ site.data.connect.core-objects.sources.retrieve.name | flatify | replace: "{source_id",source-id | remove: right-bracket | strip_newlines }}
+               -H 'Content-Type: application/json' \
+               -H 'Authorization: Bearer <API_TOKEN>'
+          ```
+
+          The response will be the source's [`report_card` object]({{ site.data.connect.api.section | flatify | prepend: site.baseurl | append: site.data.connect.data-structures.report-cards.source.section }}). In this example, the `current_step` is `4`, which corresponds to the `field_selection` step:
+
+          ```json
+          {{ site.data.connect.code-examples.source-report-cards.shopify | replace: "<STEP_NUMBER>","4" }}
+          ```
```

Comment:
fully_configured sources can also have selections altered. And (currently) in all cases field_selection and fully_configured are the last two steps when a source support field selections.

Reply:
Also a good point! I'll add that in, too.